NBitcoin/Protocol/Node.cs

```diff
@@ -149,81 +150,89 @@ public NodeConnection(Node node, Socket socket)
 			}
 
 			internal BlockingCollection<SentMessage> Messages = new BlockingCollection<SentMessage>(new ConcurrentQueue<SentMessage>());
+
 			public void BeginListen()
 			{
 				new Thread(() =>
 				{
 					SentMessage processing = null;
 					Exception unhandledException = null;
+
 					bool isVerbose = NodeServerTrace.Trace.Switch.ShouldTrace(TraceEventType.Verbose);
-					ManualResetEvent ar = new ManualResetEvent(false);
-					SocketAsyncEventArgs evt = new SocketAsyncEventArgs();
-					evt.SocketFlags = SocketFlags.None;
-					evt.Completed += (a, b) =>
-					{
-						Utils.SafeSet(ar);
-					};
-					try
 
+					using (var completedEvent = new ManualResetEvent(false))
 					{
-						foreach(var kv in Messages.GetConsumingEnumerable(Cancel.Token))
+						using (var socketEvent = NodeSocketEventManager.Create(completedEvent))

```

Comment:
Better call this socketEvent `nodeSocketEventManager` or `socketEventManager` to avoid having socketEvent.SocketEvent.

Reply:
Done!